File: VB/06_Annotations/GetAllAnnotationsFromPage.vb
Imports Spire.Pdf
Imports System.IO
Imports System.Text
Imports Spire.Pdf.Annotations

Namespace GetAllAnnotationsFromPage
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object
            Dim pdf As New PdfDocument()

            ' Load a PDF file from the specified path
            pdf.LoadFromFile("..\..\..\..\..\..\Data\Template_Pdf_3.pdf")

            ' Get the collection of annotations on the first page
            Dim annotations As PdfAnnotationCollection = pdf.Pages(0).Annotations

            ' Create a StringBuilder to store the annotation information
            Dim content As New StringBuilder()

            ' Iterate through each annotation in the collection
            For i As Integer = 0 To annotations.Count - 1

                ' Check if the current annotation is a PopupAnnotationWidget
                If TypeOf annotations(i) Is PdfPopupAnnotationWidget Then

                    ' Skip the iteration if it is a PopupAnnotationWidget
                    Continue For
                End If

                ' Append annotation information header
                content.AppendLine("Annotation information: ")

                ' Append the annotation text
                content.AppendLine("Text: " & annotations(i).Text)

                ' Get the modified date of the annotation and append it
                Dim modifiedDate As String = annotations(i).ModifiedDate.ToString()
                content.AppendLine("ModifiedDate: " & modifiedDate)
            Next i

            ' Specify the result file name
            Dim result As String = "Result-GetAllAnnotationsFromPage.txt"

            ' Write the annotation information to the result file
            File.WriteAllText(result, content.ToString())

            ' Close the PDF document
            pdf.Close()

            ' Launch the file
            PDFDocumentViewer(result)
        End Sub

        Private Sub PDFDocumentViewer(ByVal filename As String)
            Try
                Process.Start(filename)
            Catch
            End Try
        End Sub
    End Class
End Namespace

File: CS/06_Annotations/GetAllAnnotationsFromPage.cs
using System;
using System.Windows.Forms;
using Spire.Pdf;
using System.IO;
using System.Text;
using Spire.Pdf.Annotations;

namespace GetAllAnnotationsFromPage
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {

            //Create a new PDF document.
            PdfDocument pdf = new PdfDocument();

            //Load the file from disk.
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Pdf_3.pdf");

            //Get all annotations from the first page.
            PdfAnnotationCollection annotations = pdf.Pages[0].Annotations;

            StringBuilder content = new StringBuilder();

            for (int i = 0; i < annotations.Count; i++)
            {
                //A text annotation will attach a popup annotation since they are father-son relationship. 
                //The annotation information exists in the text annotation, so here we mask the blank popup annotation.
                if (annotations[i] is PdfPopupAnnotationWidget)
                    continue;
                content.AppendLine("Annotation information: ");
                content.AppendLine("Text: " + annotations[i].Text);
                string modifiedDate = annotations[i].ModifiedDate.ToString();
                content.AppendLine("ModifiedDate: " + modifiedDate);
            }

            String result = "Result-GetAllAnnotationsFromPage.txt";

            //Save to file.
            File.WriteAllText(result, content.ToString());

            //Launch the file.
            PDFDocumentViewer(result);
        }

        private void PDFDocumentViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}
